app/controllers/form_controller.rb

```diff
@@ -419,7 +420,7 @@ def render_check_errors_page
       @log.valid?
       @log.reload
       error_attributes = @log.errors.map(&:attribute)
-      @questions = @log.form.questions.select { |q| error_attributes.include?(q.id.to_sym) }
+      @questions = @log.form.questions.select { |q| error_attributes.include?(q.id.to_sym) && q.page.routed_to?(@log, current_user) }
```

Comment:
Sometimes questions have 2 versions defined, but only of of them routed to, and we end up showing 2 rows of the same question on compound validations page